### PR TITLE
Gnomonic projection using spherical trigonometry

### PR DESCRIPTION
**Pull request by astronasutarou**
The Gnomonic projection is rewritten using Spherical trigonometry.

The previous version of Gnomonic projection in `analysis/projection` used `jnp.arctan2(x, -y)` and `1/jnp.sqrt(1-z**2)`. These calculations may be problematic when x ~ 0, y ~ 0, or z ~ 1.0. The current code works better in short distance cases.

### CODE REVIEW

**Comment by astronasutarou** [comment]
Now, `astropy.wcs.WCS` is used in validation tests. The current code works in accord with `astropy.wcs.WCS` with an accuracy of 10⁻⁴ for short separation cases. I'm afraid that the current accuracy is not sufficient for the JASMINE mission...

**Comment by astronasutarou** [comment]
The same validation test was applied to the previous implementation. It worked better in accuracy. There can be some mistakes in maths in the current implementation.
